tools/rtpsrelay/ThreadMonitor.h

```diff
@@ -0,0 +1,76 @@
+#ifndef RTPSRELAY_THREAD_MONITOR_H_
+#define RTPSRELAY_THREAD_MONITOR_H_
+
+#include <dds/DCPS/ThreadMonitor.h>
+
+#include <ace/Guard_T.h>
+#include <ace/Condition_Thread_Mutex.h>
+#include <ace/Time_Value.h>
+#include <ace/Log_Msg.h>
+#include <ace/Thread_Adapter.h>
+
+#include <map>
+#include <deque>
+#include <string>
+
+using namespace std;
+
+namespace RtpsRelay {
+
+/**
+ * @class ACE_Thread_Monitor
+ *
+ * @brief Defines the means of tracking thread utilization by measuring
+ * time spent in event handling vs idle
+ */
+  class Thread_Monitor : public OpenDDS::DCPS::Thread_Monitor
+  {
+  public:
+    Thread_Monitor (int perd = 5, size_t depth = 1);
+
+    virtual void update(UpdateMode, const char* = "");
+    void summarize(void);
+    void report_thread(ACE_thread_t key);
+
+    void report(void);
+
+    void active_monitor(void);
+    void start(void);
+    void stop(void);
+
+  protected:
+    bool running_;
+    ACE_Thread_Adapter *thr_func_;
+
+    ACE_Thread_Mutex modlock_;
+    ACE_Condition_Thread_Mutex moderator_;
+
+    struct Sample {
+      UpdateMode mode_;
+      ACE_Time_Value at_;
```

Comment:
`ACE_Time_Value` doesn't indicate what it's measuring against. [Please avoid using `ACE_Time_Value` unless interacting directly with ACE.](https://opendds.readthedocs.io/en/master/internal/dev_guidelines.html#time)